### DISCUSSION TITLE
grub repair dual boot after UEFI Update

### DISCUSSION THREAD

**Discussion by santalaus45** [SpiralLinux/SpiralLinux-project#455] [General]
Hi,
i did a uefi update on my Framework 13, after this win11 is directly booted, no more grub.
The debian entry in EFI partition is still there. How did I get this repaired.

**Comment by geckolinux** [comment]
Usually the UEFI settings let you change the priority of the bootloader entries.

**Reply by santalaus45** [reply]
Sure, but it seemed that the Update deleted the NVRAM Entrys only the windows one was left.

So I need the command to reenter this entry from LiveCD. I know no Spirallinux or debian problem.

**Reply by geckolinux** [reply]
Normally there should be a FAT32 partition with an `EFI` folder, and inside that another folder for each OS bootloader, such as `debian` and `Microsoft`. If the `debian` folder is still present then your UEFI settings for the laptop should let you set Debian as the default bootloader.

**Reply by santalaus45** [reply]
Yes, the debian folder exists. When pressing F3 during boot on the Framework I could change in the folders an choose the boot loader then spirallinux will boot.
update-grub did not change anything on the boot sequence at normal startup.

**Reply by santalaus45** [reply]
found howto for grub-efi reinstall.I will try this.

**Reply by geckolinux** [reply]
Hi there, you don't need to reinstall anything. You just need to change the EFI bootloader priority:
https://community.frame.work/t/changing-efi-boot-order/47506

**Reply by santalaus45** [reply]
BTDT
Did not work since there is just the Windows entry in UEFI
The debian entry was deleted in NVRAM with the Update. will try grub-efi later and come back.
